src/selectors/Entities/nameNote.js

```diff
@@ -11,3 +13,51 @@ export const selectNameNoteIsValid = state => {
   const { isValid = false } = nameNoteState ?? {};
   return isValid;
 };
+
+export const selectClosestPCDToVaccination = (patient, vaccinationEntryDate) => {
+  const [pcdEvent] = UIDatabase.objects('PCDEvents');
+  if (!pcdEvent) return null;
+
+  const { id: pcdEventID } = pcdEvent;
+  const { nameNotes = [] } = patient ?? {};
+
+  if (!nameNotes.length) return null;
+
+  // Check first for a nameNote entered before the vaccination date
+  const filtered = nameNotes.filter(
+    ({ patientEventID, entryDate }) =>
+      patientEventID === pcdEventID && moment(vaccinationEntryDate) >= moment(entryDate)
+  );
+
+  if (!filtered.length) return null;
+
+  const sorted = filtered.sort(
+    ({ entryDate: entryDateA }, { entryDate: entryDateB }) =>
+      moment(entryDateB).valueOf() - moment(entryDateA).valueOf()
+  );
+
+  const [mostRecentPCD] = sorted;
+  return mostRecentPCD;
+};
+
+export const selectMostRecentPCD = patient => {
+  const [pcdEvent] = UIDatabase.objects('PCDEvents');
+  if (!pcdEvent) return null;
+
+  const { id: pcdEventID } = pcdEvent;
+  const { nameNotes = [] } = patient ?? {};
```

Comment:
haven't seen this before is this equivalent to: const { nameNotes } = patient ?? { nameNotes: [] };

Reply:
I think that it behaves more like: 
`const nameNotes = patient?.nameNotes ?? [];` - i.e if patient is defined but doesn't have a `nameNotes` property, would still default to empty array.

Logic was originally copied from `selectMostRecentPCD`...but on re-reading I think this is probably less confusing than `const { nameNotes = [] } = patient ?? {}; ` ... not sure though, what do you think?

Reply:
> Unclear to me is why the associated PCD name_note record is the closest one to the vaccination? is it guaranteed that each vaccination has a name_note record that is earlier than the vaccination event? For example, is it possible a patient got a booster but the name_note is from the first shot?

The UI flow to dispense a vaccination (if setup correctly) requires that a PCD name_note is filled before a vaccine can be dispensed. If, when dispensing a vaccine, there already exists a PCD `name_note` for the patient and it is not edited during the vaccine dispensing flow, no new name note will be created at the time - so the intention here is to just go back until we can find the latest... 

> Also why display the closest and not the latest name_note?
There are some things on the PCD that can change with time, e.g. pregnancy, currently breastfeeding. Kiribati would like to know what the patient's 'state' was at the time of the vaccination (e.g. they might have been pregnant at the time of vaccination, but then came in for a second dose later and were no longer pregnant)...

The overall design is ... a bit confusing. Adam has put together a slide deck to help explain the use case, might be helpful to get a picture of all the different name notes? https://docs.google.com/presentation/d/1rS0hbZTa7iAnBIGWtForcLuBiInQiGEHM33HNHsS9kI/edit#slide=id.g1197003ea89_0_564

Reply:
> I think that it behaves more like: `const nameNotes = patient?.nameNotes ?? [];` - i.e if patient is defined but doesn't have a `nameNotes` property, would still default to empty array.
> 
> Logic was originally copied from `selectMostRecentPCD`...but on re-reading I think this is probably less confusing than `const { nameNotes = [] } = patient ?? {}; ` ... not sure though, what do you think?

thanks yes `const nameNotes = patient?.nameNotes ?? [];` seems much cleaner and straight forward to me. (but maybe just because I didn't know about the other syntax)